### PR TITLE
SRCH-1605, SRCH-3636 enhance super admin reindexing tools

### PR DESCRIPTION
**Pull request by MothOnMars**
## Summary
This PR includes changes for two stories that are being tested in tandem, as each story facilitates the testing of the other:
- [SRCH-1605 allow super admins to reindex a list of URLs](https://github.com/GSA/search-gov/pull/1066/commits/b7ed8f1ee727da5bff73d766f1fcd6ca9d696710)
[SRCH-3636 allow super admins to view and search by enqueued_for_reindex](https://github.com/GSA/search-gov/pull/1066/commits/7f05101adbd16dd975803424b86dae023ce6fb70)
 
### Checklist
Please ensure you have addressed all concerns below before marking a PR "ready for review" or before requesting a re-review. If you cannot complete an item below, replace the checkbox with the ⚠️ `:warning:` emoji and explain why the step was not completed.
 
#### Functionality Checks
 
- [x] You have merged the latest changes from the target branch (usually `main`) into your branch.
  
- [x] Your primary commit message is of the format **SRCH-#### \<description\>** matching the associated Jira ticket.

- [x] PR title is either of the format **SRCH-#### \<description\>** matching the associated Jira ticket (i.e. "SRCH-123 implement feature X"), or **Release - SRCH-####, SRCH-####, SRCH-####** matching the Jira ticket numbers in the release.
 
- ⚠️  Automated checks pass. If Code Climate checks do not pass, explain reason for failures: CodeClimate is currently not running on PRs: https://status.codeclimate.com/
 
#### Process Checks

- [x] You have specified at least one "Reviewer".

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -143,6 +143,9 @@ gem 'dogapi', '~> 1.45'
 # https://github.com/ruby/net-protocol/issues/10
 # This gem can be removed once we upgrade to Ruby 3.1.
 gem 'net-http'
+# Temporarily locking the 'mail' version until the next version of Rails is released
+# https://github.com/rails/rails/pull/46650
+gem 'mail', '~> 2.7.1'
```
Comment (MothOnMars) [inline Gemfile:148]: locked to avoid another login failure

File: app/views/bulk_url_upload_results_mailer/results_email.html.erb [hunk 2]
```diff
@@ -13,7 +13,7 @@
   </p>
 <% else %>
   <p>
-    <%= @results.ok_count %> URLs were created successfully.
+    <%= @results.ok_count %> URLs were created or enqueued successfully.
```
Comment (MothOnMars) [inline app/views/bulk_url_upload_results_mailer/results_email.html.erb:16]: If necessary, a future story can list the separate numbers for created vs. enqueued, but that is out of scope for these stories.

File: db/migrate/20221213161648_add_index_on_searchgov_urls_searchgov_domain_and_enqueued_for_reindex.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+class AddIndexOnSearchgovUrlsSearchgovDomainAndEnqueuedForReindex < ActiveRecord::Migration[7.0]
+  def change
+    add_index :searchgov_urls,
```
Comment (MothOnMars) [inline db/migrate/20221213161648_add_index_on_searchgov_urls_searchgov_domain_and_enqueued_for_reindex.rb:3]: added to support super admin search by `enqueued_for_reindex`

File: features/admin.feature [hunk 4]
```diff
@@ -177,9 +177,6 @@ Feature:  Administration
       | domain     | status | canonical_domain |
       | search.gov | 200 OK |                  |
       | old.gov    | 200 OK | new.gov          |
-    And the following "searchgov urls" exist:
```
Comment (MothOnMars) [inline features/admin.feature:180]: The `Managing Search.gov Domains` scenario was getting rather unwieldy, so I've separated it into `Managing Search.gov Domains` and `Managing Search.gov URLs`. (Eventually I'd like to move each super admin page to its own feature file, but refactoring can wait.)

File: spec/system/admin/bulk_url_upload_spec.rb [hunk 5]
```diff
@@ -16,9 +16,10 @@
   include ActiveJob::TestHelper
 
   subject(:do_bulk_upload) do
-    perform_enqueued_jobs do
+    perform_enqueued_jobs except: SearchgovDomainIndexerJob do
```
Comment (MothOnMars) [inline spec/system/admin/bulk_url_upload_spec.rb:19]: This ensures we run the jobs associated with the bulk upload functionality, but not actually start indexing the URLs, as that causes unrelated spec failures.

File: Gemfile (hunk 1 above)
Comment (ddzz) [reply Gemfile:148]: Should this be moved above the `Assets-related gems` section?

File: spec/system/admin/bulk_url_upload_spec.rb [hunk 6]
```diff
@@ -66,6 +68,23 @@
...
+        let(:reindex) { true }
+        let!(:existing_url) do
+          SearchgovUrl.create!(
+            url: 'https://agency.gov/',
+            last_crawled_at: 1.week.ago,
+            last_crawl_status: 'OK'
+          )
+        end
+
+        it 'reindexes the URLs' do
```
Comment (ddzz) [inline spec/system/admin/bulk_url_upload_spec.rb:82]: Can an assertion be added here to check that reindexing is triggered for the URL, or is that not possible with the `except` in L19?

File: Gemfile (hunk 1 above)
Comment (MothOnMars) [reply Gemfile:148]: No, the `mail` gem is not related to assets: https://github.com/mikel/mail

**Review by ddzz — APPROVED** [review]
LGTM

**Pull request merged**